src/main/java/seedu/address/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.EMPTY_PREFIX;
+
+import java.util.List;
+
+import seedu.address.logic.commands.FilterCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.category.Category;
+
+/**
+ * Parses input arguments and creates a new {@code FilterCommand} object
+ */
+public class FilterCommandParser implements Parser<FilterCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code FilterCommand}
+     * and returns a {@code FilterCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        String trimmedArgs = args.trim();
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(trimmedArgs, EMPTY_PREFIX);
+
+        Category category;
+        Integer count;
+        List<String> argList = argMultimap.getAllValues(EMPTY_PREFIX);
+
+        if (!areArgumentsPresent(argList)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+        }
+
+        if (areBothArgumentsPresent(argList)) {
+            // category and count arguments are specified
+            category = extractCategoryForFilter(argList);
+            count = extractCountForFilter(argList);
+        } else {
+            // only category argument specified
+            category = extractCategoryForFilter(argList);
+            count = Integer.MAX_VALUE;
+        }
+
+        return new FilterCommand(category, count);
+    }
+
+    /**
+     * Returns true if all the arguments are present in the given
+     * {@code List<String>}.
+     */
+    private static boolean areArgumentsPresent(List<String> argList) {
+        if (argList == null) {
+            return false;
+        }
+        return argList.size() == 1 || argList.size() == 2;
```

Comment:
What happens if the user enters filter gender name 1. Wouldnt that return false even tho there are technically arguments provided? The add function for example only accepts the last valid input for a field. So if lets say i try to add 2 names, it will take the second one. Do you want to try implementing a functionality like that? So in the example i gave at the start, the filter function will filter by name.